### PR TITLE
Empty links warn

### LINKED ISSUE #27: TODO headline degeneracy cause nil string match error
```org
* TODO
  There is no text in this headline, and TOC update will fail.  Ignoring is fine.  This is degenerate.
```

### PR DESCRIPTION
**Pull request by psionic-k**
Normalizing away nils in the 4th header component results in org warning the user that they have an empty link.  The TOC does not finish generation in this case.

At first I just fixed the string-match-p call, but this will complete without generating user feedback about the empty title.

fixes #27 

Signed-off-by: Psionik K <73710933+psionic-k@users.noreply.github.com>

### CODE REVIEW

**Comment by alphapapa** [comment]
I appreciate the PR, but with regard to #27, it would be helpful if we decided what to do in plain English before deciding how to change the code.  If you'd like, please discuss some ideas there first.

File: org-make-toc.el [hunk 1]
```diff
@@ -329,8 +329,11 @@ with the destination of the published file."
                                  (seq-intersection org-make-toc-exclude-tags (org-get-tags))
                                  ;; NOTE: The "COMMENT" keyword is not returned as the to-do keyword
                                  ;; by `org-heading-components', so it can't be tested as a keyword.
-                                 (when-let ((headline-text ((nth 4 (org-heading-components)))))
-                                   (string-match-p (rx bos "COMMENT" (or blank eos)) headline-text)))
+                                 (if-let ((headline-text (nth 4 (org-heading-components))))
+                                     (string-match-p (rx bos "COMMENT" (or blank eos)) headline-text)
+                                   (delay-warning 'org-make-toc
```
Comment (psionic-k) [inline org-make-toc.el:334]: I didn't understand the manual or docstring's explanation of the TYPE argument to `delay-warning`, but use-package macros expand such that they use the package name symbol when calling `display-warning`.  As far as I can tell, any symbol is accepted, and the output to the user looks informative when done this way.
